src/utils/fetchClient.ts

```diff
@@ -1,39 +1,36 @@
-const BASE_URL = 'https://mate.academy/students-api';
+import { Comment } from '../types/Comment';
 
-// a promise resolved after a given delay
-function wait(delay: number) {
-  return new Promise(resolve => {
-    setTimeout(resolve, delay);
-  });
-}
+const BASE_URL = 'https://mate.academy/students-api';
 
-// To have autocompletion and avoid mistypes
 type RequestMethod = 'GET' | 'POST' | 'PATCH' | 'DELETE';
 
 function request<T>(
   url: string,
   method: RequestMethod = 'GET',
-  data: any = null, // we can send any data to the server
+  data: Comment | null = null,
 ): Promise<T> {
   const options: RequestInit = { method };
 
   if (data) {
-    // We add body and Content-Type only for the requests with data
     options.body = JSON.stringify(data);
     options.headers = {
       'Content-Type': 'application/json; charset=UTF-8',
     };
   }
 
-  // for a demo purpose we emulate a delay to see if Loaders work
-  return wait(300)
-    .then(() => fetch(BASE_URL + url, options))
-    .then(response => response.json());
+  return fetch(BASE_URL + url, options)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Unable to connect from the server');
```

Comment:
The error message 'Unable to connect from the server' may not accurately represent the error. Consider using response status and status text for more accurate error messages.